app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/AlertPlayer.java

```diff
@@ -0,0 +1,130 @@
+        Log.e(TAG, "Snooze called repeatTime = "+ repeatTime);
+        stopAlert(false, false);
+        ActiveBgAlert activeBgAlert = ActiveBgAlert.getOnly();
+        if (activeBgAlert  == null) {
+            Log.e(TAG, "Error, snooze was called but no alert is active. how can that be ??? !!! ");
+        }
+        activeBgAlert.snooze(repeatTime);
+    }
+    
+ // Check the state and alrarm if needed
+    public void ClockTick(Context ctx)
+    {
+        ActiveBgAlert activeBgAlert = ActiveBgAlert.getOnly();
+        if (activeBgAlert  == null) {
+            // Nothing to do ...
+            return;
+        }
+        if(activeBgAlert.ready_to_alarm()) {
+            stopAlert(false, false);
+            AlertType alert = AlertType.get_alert(activeBgAlert.alert_uuid);
+            if (alert == null) {
+                Log.w(TAG, "ClockTick: The alert was already deleted... will not play");
+                ActiveBgAlert.ClearData();
+                return;
+            }
+            Log.e(TAG,"ClockTick: Playing the alert again"); 
+            PlayFile(ctx, alert.mp3_file);
+        }
+        
+    }
+
+    private void PlayFile(Context ctx, String FileName) {
+        if(mediaPlayer != null) {
+            Log.e(TAG, "ERROR, PlayFile:going to leak a mediaplayer !!!");
+        }
+        if(FileName != null) {
+            mediaPlayer = MediaPlayer.create(ctx, Uri.parse(FileName), null);
+        }
+        if(mediaPlayer == null) {
+            Log.w(TAG, "PlayFile: Creating mediaplayer with file " + FileName + " failed. using default alarm");
+            mediaPlayer = MediaPlayer.create(ctx, R.raw.default_alert);
+        }
+        if(mediaPlayer != null) {
+            
+            AudioManager manager = (AudioManager) ctx.getSystemService(Context.AUDIO_SERVICE);
+            int maxVolume = manager.getStreamMaxVolume(AudioManager.STREAM_MUSIC);
+            volumeBeforeAlert = manager.getStreamVolume(AudioManager.STREAM_MUSIC);
+            manager.setStreamVolume(AudioManager.STREAM_MUSIC, maxVolume, 0);
+            context = ctx;
+
+            mediaPlayer.setOnCompletionListener(new MediaPlayer.OnCompletionListener() {
+                @Override
+                public void onCompletion(MediaPlayer mp) {
+                    AudioManager manager = (AudioManager) context.getSystemService(Context.AUDIO_SERVICE);
+                    int currentVolume = manager.getStreamVolume(AudioManager.STREAM_MUSIC);
+                    int maxVolume = manager.getStreamMaxVolume(AudioManager.STREAM_MUSIC);
+                    if(maxVolume == currentVolume) {
+                        // If the user has changed the volume, don't change it again.
+                        manager.setStreamVolume(AudioManager.STREAM_MUSIC, volumeBeforeAlert, 0);
+                    }
+                }
+            });
+            
+            mediaPlayer.start();
+        } else {
+            // TODO, what should we do here???
```

Comment:
Should we maybe just try to raise a normal notification with audio? Its not ideal but probably better than nothing... Do you know if there is a way to change notification volume using the same settings permission?